log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/RollingRandomAccessFileManager.java

```diff
@@ -174,8 +174,10 @@ public static RollingRandomAccessFileManager getRollingRandomAccessFileManager(
                             long size = 0;
                             long time = System.currentTimeMillis();
                             RandomAccessFile raf = null;
+                            boolean fileExistedBefore = false;
```

Comment:
Can we replace the `file.exists()` on line 184 with `fileExistedBefore` too?

Reply:
no, it should be re-evaluate to make sure time is not based on the one deleted